### PR TITLE
[Feat] added slog to printed errors and some info logging

### LINKED ISSUE #1: Convert fmt print statements to use a logger instead

### PR DESCRIPTION
**Pull request by roceb**
Fixes #1 
Replaced fmt.print with slog

### CODE REVIEW

File: controllers/quote.go [hunk 1]
```diff
@@ -54,6 +55,7 @@ func GetQuotes(ctx *gin.Context) {
 	}
 
 	if err != nil {
+		slog.Error("Something went wrong", "error", err)
```
Comment (jeyren95) [inline controllers/quote.go]: we can remove this since the errors caused by querying the DB have already been logged

File: queries/quote.go [hunk 2]
```diff
@@ -27,15 +26,13 @@ func scanRows(rows *sql.Rows) ([]*models.Quote, error) {
 
 func InsertQuote(quote models.Quote) error {
 	statement, err := db.DB.Prepare("INSERT INTO quote(quote, character, season, episode, title) VALUES($1, $2, $3, $4, $5)")
-
 	if err != nil {
-		fmt.Print(err)
+		slog.Error(err)
```
Comment (jeyren95) [inline queries/quote.go]: can we make this error log more verbose? maybe something like this

```suggestion
		slog.Error("InsertQuote", "prepare SQL statement", err)
```

File: queries/quote.go [hunk 3]
```diff
@@ -27,15 +26,13 @@ func scanRows(rows *sql.Rows) ([]*models.Quote, error) {
...
 		return err
 	}
 	defer statement.Close()
 
 	_, err = statement.Exec(quote.Quote, quote.Character, quote.Season, quote.Episode, quote.Title)
-
 	if err != nil {
 		return err
```
Comment (jeyren95) [inline queries/quote.go:38]: Include logging here too

```suggestion
		return slog.Error("InsertQuote", "scan rows", err)
```

File: controllers/quote.go [hunk 4]
```diff
@@ -72,8 +74,8 @@ func GetQuotes(ctx *gin.Context) {
 //	@Failure	500	{object}	models.ErrorResponse
 func GetRandomQuote(ctx *gin.Context) {
 	count, err := queries.GetQuoteCount()
-
 	if err != nil {
+		slog.Error("Something went wrong", "error", err)
```
Comment (jeyren95) [inline controllers/quote.go:77]: remove this too

File: queries/quote.go (hunk 3 above)
Comment (roceb) [reply queries/quote.go:38]: I assume you still want to return the err

> slog.Error("InsertQuote", "scan rows", err) 
> return err

Comment (jeyren95) [reply queries/quote.go:38]: Yep, pls return the error

**Comment by roceb** [comment]
Updated code with your suggestions

⁣

File: queries/quote.go [hunk 5]
```diff
@@ -27,13 +27,14 @@ func scanRows(rows *sql.Rows) ([]*models.Quote, error) {
 func InsertQuote(quote models.Quote) error {
 	statement, err := db.DB.Prepare("INSERT INTO quote(quote, character, season, episode, title) VALUES($1, $2, $3, $4, $5)")
 	if err != nil {
-		slog.Error(err)
+		slog.Error("Something went wrong", "error", err)
```
Comment (jeyren95) [inline queries/quote.go:30]: can change this to `slog.Error("InsertQuote", "prepare SQL statement", err)`

**Comment by jeyren95** [comment]
will you be adding log statements to the other queries too?

**Review by jeyren95 — APPROVED** [review]

**Comment by jeyren95** [comment]
thanks for your help! do give any suggestions for the project if you have any

**Comment by roceb** [comment]
I added more logging. Ill send a new PR. I was also thinking about maybe switching to sqlx.

**Comment by jeyren95** [comment]
ok lemme take a look at sqlx, and i'll raise an issue to switch to it if needed. thanks for your contribution!

**Pull request merged**